### DISCUSSION TITLE
Welcome to AstroMosaic Discussions!

### DISCUSSION THREAD

**Discussion by jarmoruuth** [jarmoruuth/AstroMosaic#2] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by danielerisolo** [comment]
Hi,
great idea!
my name is Daniele and I love AstroMosaic!
as already Jarmo knows, my wish is to add also planets !
ciao a tutti!
daniele


Il giorno gio 23 mar 2023 alle ore 21:05 jarmoruuth <
***@***.***> ha scritto:

**Reply by jarmoruuth** [reply]
Hi Daniele,

Thanks for reminding! :-) I will take a look. There is an API to get planet locations at https://ssd.jpl.nasa.gov/horizons/. If that does not work the I try to do calculation myself using information in here: https://stjarnhimlen.se/comp/ppcomp.html

I guess it would be nice to see altitude like with the moon and also days path on the sky.

Jarmo

**Comment by danielerisolo** [comment]
great!
have a nice weekend!
daniele

Il giorno ven 24 mar 2023 alle 08:23 jarmoruuth ***@***.***>
ha scritto:

**Comment by danielerisolo** [comment]
Hi, I found this: https://github.com/dcf21/ephemeris-compute-de430 I don't
know if it should be useful...
ciao!

Il giorno ven 24 mar 2023 alle ore 08:23 jarmoruuth <
***@***.***> ha scritto:

**Reply by jarmoruuth** [reply]
Hi, 

Possibly that could be useful. But I just completed calculations from here: https://stjarnhimlen.se/comp/ppcomp.html

I could not access NASA site (https://ssd.jpl.nasa.gov/horizons/) from Javascript. But it can be accessed from browser Url so it can used to check calculations.

Jarmo

**Comment by jarmoruuth** [comment]
I now have a test version at https://ruuth.xyz/test/AstroMosaic.html. It can show planet daily visibility in a chart. Next it would be nice to see the path in a sky atlas (moon and planet).

**Comment by jarmoruuth** [comment]
Now also map view and year view in https://ruuth.xyz/test/AstroMosaic.html.

Jarmo

**Comment by danielerisolo** [comment]
Some test done on all planets and data matches Stellarium info!
I've only noticed that the add telescope bar is present from the beginning
and if you want to disappear it, you have to flag "add telescope"  and then
click again to unflag it.

Some little request about aladin view in the main page only  (not mosaic or
panel view):
1. is it possible to add the pointer? ( showSimbadPointerControl = true)
request for the main page control:
1. is it possible to add 35° and 40 in the altitude dropdown list?

other: are you planning to switch to aladin 3 ?

ciao :-)






Il giorno dom 26 mar 2023 alle ore 00:17 jarmoruuth <
***@***.***> ha scritto:

**Comment by jarmoruuth** [comment]
Hi,

Great that you verified planet data!

Now all those changes and fixes should be in the test version. Added some
other stuff there like Current Location "Telescope" just to show what is
above the current location (not sure if that is calculated correctly).

The tip on the Simbad pointer was very cool! I like it a lot.

I tested Aladin v3. It changed how the page looks so I need to take a
closer look at it.

Jarmo


su 26. maalisk. 2023 klo 18.33 danielerisolo ***@***.***)
kirjoitti:

**Comment by danielerisolo** [comment]
Hi,
"Current location" is a great idea! I love it
I've checked and it seems to be correct:

[image: image.png]
But I miss the wheater location, isn't it?

other question:
I've noticed a "circle" that appear at the end of the moon and planets
track, if you hover it with mouse what can you do?
[image: image.png]


Il giorno dom 26 mar 2023 alle ore 21:11 jarmoruuth <
***@***.***> ha scritto:

**Comment by jarmoruuth** [comment]
Hi,

Good point, I forgot the weather. There is "detect location" in the weather app so it should be easy.

The circle is just supposed to show the start location of the path. So you know which direction it is moving. It would be nice to hover over the path and see the date and time at that location. I think it should be possible.

Jarmo

**Comment by danielerisolo** [comment]
Great, I'd like to ask you for hover over the path and see the date and
time at that location, but I didn't dare... :-)


Il giorno lun 27 mar 2023 alle ore 09:25 jarmoruuth <
***@***.***> ha scritto:

**Comment by jarmoruuth** [comment]
Added dome hovering :-)

Jarmo

**Comment by danielerisolo** [comment]
*Great*!
I've noticed that for moon , every blue circle is 1 hour time, while for
planet (pink) every circle is 1 day, is it correct?
[image: image.png]

Il giorno lun 27 mar 2023 alle ore 20:42 jarmoruuth <
***@***.***> ha scritto:

**Comment by jarmoruuth** [comment]
Yes, moon is shown for 24 hours, every hour. Planets move so slowly so they are printed for the next 4 weeks (28 days).

Not sure what is most useful, these could of course be configurable. 

Also it could be possible to show multiple planets. I am running out of space on the screen so this would likely need a new line added to the interface.

Jarmo

**Reply by danielerisolo** [reply]
Ok, for the main use that I make of it, i.e. the programming of an observation with the telescope, it would be more convenient to view it every hour rather than every 24 hours. But this is only my opinion, based on my type of use. For a "generic" use can be okay every 1 day.
I've probably found an issue with Simbad catalog, I'll write in a separate post :-)

**Reply by danielerisolo** [reply]
Uh, I forget: **I love** that when you flag on some catalogs (IC NGC Messier) and I hover it I get the info about the object!!! 
![image](https://user-images.githubusercontent.com/104658608/228208198-1995c4d4-2d4f-4694-ae6f-f8e6a8a659ed.png)

**Comment by jarmoruuth** [comment]
True, once a day is not good for finding the position. Looks like you cannot hover over a line so I added those "catalog" objects (circles). Maybe adding a smaller circle on every hour would be ok. But there is not much of movement in one hour. There could also be a click action that would show a popup with a list of more precise time and location. Let me know if you have ideas, otherwise I do just do something and we can improve from there :-)

Jarmo

**Reply by danielerisolo** [reply]
I think your idea of a popup window is great!

**Reply by danielerisolo** [reply]
There would be many ideas, but I wouldn't want to go beyond the scope you created AstroMosaic for. I don't want a double of Stellarium or anything like that. maybe I'm exaggerating with the requests! Give me some time to use it "in the field" (weather permitting), so I can better understand what can really be useful. What do you think?

**Reply by jarmoruuth** [reply]
I agree, there a lot of great tools like Stellarium so no point of trying to do too much. But I like to see the moon and planet path and visibility so those are fun and useful additions. I can do some small stuff like the popup or similar since the data is already available.

Jarmo

**Comment by danielerisolo** [comment]
Question about **Simbad** catalog:
this is what I'm doing:
1. start AstroMosaic: default: Telescope Live - SPA-2-CMOS target: M74
2. flag ON the Simbad Catalog (that is only active on the actual FoV, correct?) - I see a numbers of objects:
![image](https://user-images.githubusercontent.com/104658608/228209322-33717f95-3064-49d9-9b66-6b0114784706.png)
3. Click on "Reposition" flag
![image](https://user-images.githubusercontent.com/104658608/228209501-ee2c3eb4-3527-426a-bf3e-11974d198ac2.png)
4. If Move the sky with mouse, I lost the Simbad catalog:
![image](https://user-images.githubusercontent.com/104658608/228209882-01242d07-148d-45eb-9768-76dafaec928a.png)

Do you know why?

PS: when I hover the mouse over a Simbad Object, I don't get the information text like for other catalogs/planet
I hope I explained myself!

daniele

**Reply by jarmoruuth** [reply]
Looks like Reposition looses catalogs, need to look into it.

Hovering works only if there are known fields in the data. Need to check what is the data format for Simbad objects. If it is easy to understand it is also easy to add.

Jarmo

**Comment by danielerisolo** [comment]
Hi I've probably found a little issue with Time Zone calculation using "Current Location"
I'm in Italy, so what I'm expecting to see is TimeZone +2 , but I get "-2":
![image](https://user-images.githubusercontent.com/104658608/228243584-dda0b529-0fda-415f-9a89-ebbb85ce7f69.png)

**Comment by jarmoruuth** [comment]
Most of the problems should be resolved. Hopefully not too many added :-)

Hover, Click, reposition and catalogs

- Do not move Simbad and NED catalogs when repositioning. System seem to lost track with that. Click Refresh and show catalog again to show it in a new location.
- Updates to hovering and click listeners. More objects are shown. Clicking will show info table at the bottom of the page.
- When clicking moon or planet circle a more detailed positions are shows up to next circle.
- Fix to time zone offset.

Jarmo

**Comment by danielerisolo** [comment]
v1.74 tested with: **M35 - Moon and Mars:**
![AM-M35-Mars-Moon](https://user-images.githubusercontent.com/104658608/228365365-d717afa1-c6a8-4be4-b96f-2776593f1e97.PNG)

M35 (green line in the above picture) is less than **less than 2°** from Mars (pink line in the above picture, practically overlapped): **CONFIRMED!**
![Marte-M35](https://user-images.githubusercontent.com/104658608/228365501-16ab18bb-b574-4416-ab5d-c435a220c8ab.PNG)

Mars and moon are at **less than 4°**: **CONFIRMED!**
![Marte-Luna](https://user-images.githubusercontent.com/104658608/228365700-a2edfb66-e747-42f4-baf3-3c8aafaaf349.PNG)

**Comment by danielerisolo** [comment]
Random REAL test with telescope: **NGC2174** (**Monkey Head Nebula**)
Check for elevation (azimut): **46° CONFIRMED!!**!
![MonkeyNebula](https://user-images.githubusercontent.com/104658608/228366420-643b622b-561e-40ea-80f3-306467494255.PNG)

I think that v1.74 should be released on Github!
ciao
Daniele

**Comment by jarmoruuth** [comment]
Great, thanks for testing!

I would like to add you to the credits here; https://ruuth.xyz/AstroMosaicInfo.html. If it is ok, what name should I use?

Jarmo

**Comment by danielerisolo** [comment]
Wow, this is an honor!
my name is Daniele Risolo !
grazie!


Il giorno mer 29 mar 2023 alle 10:54 jarmoruuth ***@***.***>
ha scritto:

**Comment by jarmoruuth** [comment]
Found a bug and updated the current location service. Now you can use also time, and there is telescope Now for automatic current time. Updated the info page, changes including credits are here: https://ruuth.xyz/test/AstroMosaicInfo.html

I hope it is done now but need do some more tests.

Jarmo

**Comment by danielerisolo** [comment]
thanks a lot!
I’ll do my best to check every possible issues
ciao ✋

Il giorno mer 29 mar 2023 alle 23:20 jarmoruuth ***@***.***>
ha scritto:

**Comment by jarmoruuth** [comment]
I finally put out v1.74.

Jarmo

**Comment by danielerisolo** [comment]
Great,
my tests go ahead, I let you know in case of little issues
ciao
Daniele

Il giorno dom 2 apr 2023 alle ore 22:43 jarmoruuth ***@***.***>
ha scritto:

**Comment by danielerisolo** [comment]
Hello,
I’ve noticed that for every telescope you select, the Time Zone is always +2
is this a bug?
Sorry but my previous tests were focused on my own “added” telescopes, so I
didn’t checked it

I take the occasion to wish you a nice Easter!


Il giorno dom 2 apr 2023 alle 22:43 jarmoruuth ***@***.***>
ha scritto:

**Comment by jarmoruuth** [comment]
Hi, Thanks for noticing this! I also missed the problem. A fixed version 1.74.1 is now pushed.

A Happy Easter for you too!

Jarmo

**Comment by danielerisolo** [comment]
Hi, I'm testing v1.74.2, I've noticed an issue with weather location.
I'm using a virtual machine located in UK, if I use "current location" I correctly get as weather forecast "London".
But if I add  a telescope, and I enter my coordinates for Italy, I still get meteo for London and not Milan as expected.
I seems that act as if I use "get location" button, but is not the case...
any idea about it?
ciao! 
Daniele

**Comment by jarmoruuth** [comment]
Hi, Unfortunately the site I am using for weather widgets does not support location by coordinates. At least I have not found a way do that. So the only options are to let it detect location or find your location in https://www.meteoblue.com/en/weather/widget/setupseeing/princeton_united-states_5102922 and use the generated link. Right now there is no other way to give the weather link than modifying the source code. How do you have add a telescope, using the add button?

Jarmo

**Comment by danielerisolo** [comment]
Yes, I’ve used the “Add” flag


Il giorno gio 13 apr 2023 alle 17:41 jarmoruuth ***@***.***>
ha scritto:

**Comment by jarmoruuth** [comment]
I have a new test version at https://ruuth.xyz/test/AstroMosaic.html.

In that version weather shows the weather forecast from Windy. New seeing shows the old weather info. Seeing works like the old weather, it cannot use telescope coordinates. New weather always uses telescope coordinates. You can choose either one or both.

I kind of like it although Windy is a bit slow on updating detailed info.

Jarmo

**Comment by danielerisolo** [comment]
I like it too!
visually I think is better
It’s nice to have
great job!

Il giorno gio 13 apr 2023 alle 23:16 jarmoruuth ***@***.***>
ha scritto:

**Comment by danielerisolo** [comment]
Just tested from my VPS machine (UK), telescope added with Italy
coordinates: weather is OK (showing Milan), and seeing meteoblu is still
showing UK location
[image: image.png]

Il giorno gio 13 apr 2023 alle ore 23:16 jarmoruuth <
***@***.***> ha scritto:

**Comment by jarmoruuth** [comment]
Good that the new weather works. I could not figure out how to make seeing work with coordinates.

Jarmo
